kof-operator/internal/server/handlers/handlers.go

```diff
@@ -0,0 +1,160 @@
+package handlers
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/fs"
+	"net/http"
+	"path"
+	"strings"
+
+	"github.com/k0rdent/kof/kof-operator/internal/k8s"
+	"github.com/k0rdent/kof/kof-operator/internal/models/target"
+	"github.com/k0rdent/kof/kof-operator/internal/server"
+	static "github.com/k0rdent/kof/kof-operator/webapp/collector"
+)
+
+const MothershipClusterName = "mothership"
+
+func NotFoundHandler(res *server.Response, req *http.Request) {
+	res.Writer.Header().Set("Content-Type", "text/plain")
+	res.SetStatus(http.StatusNotFound)
+	_, err := fmt.Fprintln(res.Writer, "404 - Page not found")
+	if err != nil {
+		res.Logger.Error(err, "Cannot write response")
+	}
+}
+
+func ReactAppHandler(res *server.Response, req *http.Request) {
+	if serveStaticFile(res, req, static.ReactFS) {
+		return
+	}
+	NotFoundHandler(res, req)
+}
+
+func PrometheusHandler(res *server.Response, req *http.Request) {
+	ctx := req.Context()
+	targets := &target.PrometheusTargets{}
+
+	kubeClient, err := k8s.NewClient()
+	if err != nil {
+		res.Logger.Error(err, "Failed to create client")
+		return
+	}
+
+	cdList, err := k8s.GetClusterDeployments(ctx, kubeClient.Client)
+	if err != nil {
+		res.Logger.Error(err, "Failed to get cluster deployments")
+	}
+
+	clusters := make([]*k8s.Cluster, 0, len(cdList.Items))
```

Comment:
You don't check cdList for nil here, but in case of error you will get a panic here as cdList will be nil in case of error.